synthesizer/src/vm/helpers/cost.rs

```diff
@@ -12,17 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::VM;
+use crate::{
+    prelude::{Stack, StackProgramTypes},
+    VM,
+};
 use console::{
     prelude::*,
-    program::{LiteralType, PlaintextType},
+    program::{FinalizeType, LiteralType, PlaintextType},
 };
 use ledger_block::{Deployment, Execution};
 use ledger_store::ConsensusStorage;
-use synthesizer_program::{Command, Finalize, Instruction};
+use synthesizer_program::{CastType, Command, Finalize, Instruction, Operand, StackProgram};
 
 use std::collections::HashMap;
 
+// Finalize Costs for compute heavy operations. Used as BASE_COST + PER_BYTE_COST * SIZE_IN_BYTES.
```

Comment:
How were the numbers backed into? I realize the target was ~100 credits per second of finalize runtime, but was there a formula for determining the base cost and per byte cost in relation to the observed runtime?

Reply:
In addition, did we take into account that the scaling of runtime was linear for increases to the number of bytes of the operands? Because the fee model assumes the linear growth.

Reply:
> In addition, did we take into account that the scaling of runtime was linear for increases to the number of bytes of the operands? Because the fee model assumes the linear growth.

Looking at the data, it looks like the runtimes scale fairly linearly with the input size.

Reply:
> How were the numbers backed into? I realize the target was ~100 credits per second of finalize runtime, but was there a formula for determining the base cost and per byte cost in relation to the observed runtime?

The values were regressed (using `ordinary least squares linear regression`) over using `bytes` as the independent variable and `benchmarked runtime` as the dependent variable to give linear formulas (of the form A*bytes + C) for predicting runtime. 

The resulting equations are of the form are:
`runtime(bytes) = A*bytes + C` - both A & C were then extracted to be the Cost multiplier for the opcode & the base cost for the opcode respectively.

> In addition, did we take into account that the scaling of runtime was linear for increases to the number of bytes of the operands? Because the fee model assumes the linear growth.

If you do look at the data you will indeed find there are some outliers, but as you say it scales "mostly" linearly